assets/src/blocks/Covers/CoversBlock.js

```diff
@@ -1,130 +1,99 @@
-            default: []
-          },
-          covers_view: {
-            type: 'string',
-            default: '1'
-          },
-          cover_type: {
-            type: 'string',
-          }
-        },
-        // withSelect is a "Higher Order Component", it works as
-        // a Decorator, it will provide some basic API functionality
-        // through `select`.
-        edit: withSelect( ( select ) => {
-          const currentPostType = select('core/editor').getCurrentPostType();
+const getStyleLabel = (label, help) => {
+  if (help) {
+    return (
+      <Tooltip text={help}>
+        <span>{label}</span>
+      </Tooltip>
+    );
+  }
+  return label;
+};
 
-          return {
-            currentPostType,
-          };
-        } )( ( {
-          currentPostType,
-          isSelected,
-          attributes,
-          setAttributes
-        } ) => {
+const attributes = {
+  cover_type: {
+    type: 'string',
+    default: '3',
+  },
+  covers_view: {
+    type: 'string',
+    default: '1'
+  },
+  title: {
+    type: 'string',
+    default: '',
+  },
+  description: {
+    type: 'string',
+    default: '',
+  },
+  tags: {
+    type: 'array',
+    default: []
+  },
+  post_types: {
+    type: 'array',
+    default: []
+  },
+  posts: {
+    type: 'array',
+    default: []
+  },
+};
 
-            if ( !currentPostType ) {
-                return "Populating block's fields...";
-            }
+export const registerCoversBlock = () => {
+  const { registerBlockType } = wp.blocks;
 
-            return <Covers
-              attributes={ attributes}
-              setAttributes={ setAttributes}
-              isSelected={ isSelected }
-              currentPostType={ currentPostType }
-            />
-        } ),
+  registerBlockType(BLOCK_NAME, {
+    title: 'Covers',
+    icon: 'slides',
+    category: 'planet4-blocks',
+    supports: {
+      html: false, // Disable "Edit as HTMl" block option.
+    },
+    attributes,
+    edit: CoversEditor,
+    save: frontendRendered(BLOCK_NAME),
+    // Add our custom styles
+    styles: [
+      {
+        name: 'content',
+        label: getStyleLabel(
+          __('Content covers', 'planet4-blocks-backend'),
+          __('Content covers pull the image from the post', 'planet4-blocks-backend'),
+        ),
+        isDefault: true
+      },
+      {
+        name: 'take-action',
+        label: getStyleLabel(
+          __('Take Action covers', 'planet4-blocks-backend'),
+          __('Take action covers pull the featured image, tags, have a 25 character excerpt and have a call to action button', 'planet4-blocks-backend'),
+        ),
+      },
+      {
+        name: 'campaign',
+        label: getStyleLabel(
+          __('Campaign covers', 'planet4-blocks-backend'),
+          __('Campaign covers pull the associated image and hashtag from the system tag definitions', 'planet4-blocks-backend'),
+        ),
+      }
+    ],
+    deprecated: [
+      {
+        attributes,
```

Comment:
If we add the same `attributes` as the current version then removing, changing or adding any attribute in the current version will cause the deprecated entry to stop working as intended. If you move each deprecated entry to its own file, you can copy the attributes and they're still out of view.

Reply:
For example like we did for Media https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/wysiwyg-vars/assets/src/blocks/Media/MediaBlock.js#L56-L58